app/src/main/kotlin/io/homeassistant/companion/android/widgets/todo/TodoWidget.kt

```diff
@@ -1,456 +1,344 @@
+import androidx.glance.layout.padding
+import androidx.glance.layout.size
+import androidx.glance.preview.ExperimentalGlancePreviewApi
+import androidx.glance.preview.Preview
+import androidx.glance.semantics.semantics
+import androidx.glance.semantics.testTag
+import androidx.glance.text.Text
+import androidx.glance.text.TextAlign
+import dagger.hilt.EntryPoint
+import dagger.hilt.EntryPoints
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
 import io.homeassistant.companion.android.R
-import io.homeassistant.companion.android.common.R as RCommon
-import io.homeassistant.companion.android.common.data.integration.Entity
-import io.homeassistant.companion.android.common.data.integration.friendlyName
-import io.homeassistant.companion.android.common.data.servers.ServerManager
-import io.homeassistant.companion.android.common.data.websocket.impl.entities.GetTodosResponse
-import io.homeassistant.companion.android.database.widget.TodoWidgetDao
-import io.homeassistant.companion.android.database.widget.TodoWidgetEntity
+import io.homeassistant.companion.android.common.R as commonR
 import io.homeassistant.companion.android.database.widget.WidgetBackgroundType
-import io.homeassistant.companion.android.util.getAttribute
-import io.homeassistant.companion.android.webview.WebViewActivity
-import io.homeassistant.companion.android.widgets.BaseWidgetProvider
-import javax.inject.Inject
-import kotlinx.coroutines.launch
-import timber.log.Timber
-
-@AndroidEntryPoint
-class TodoWidget : BaseWidgetProvider() {
-
-    companion object {
-        private const val OPEN_TODO_LIST =
-            "io.homeassistant.companion.android.widgets.todo.TodoWidget.OPEN_TODO"
-        private const val TOGGLE_TODO_ITEM =
-            "io.homeassistant.companion.android.widgets.todo.TodoWidget.TOGGLE_TODO_ITEM"
-        internal const val EXTRA_SERVER_ID = "EXTRA_SERVER_ID"
-        internal const val EXTRA_ENTITY_ID = "EXTRA_ENTITY_ID"
-        internal const val EXTRA_SHOW_COMPLETED = "EXTRA_SHOW_COMPLETED"
-        internal const val EXTRA_TOGGLE_SUMMARY = "EXTRA_TOGGLE_SUMMARY"
-        internal const val EXTRA_TOGGLE_TARGET_STATUS = "EXTRA_TOGGLE_TARGET_STATUS"
-        internal const val EXTRA_BACKGROUND_TYPE = "EXTRA_BACKGROUND_TYPE"
-        internal const val EXTRA_TEXT_COLOR = "EXTRA_TEXT_COLOR"
-
-        private data class TodoWidgetData(
-            val name: String?,
-            val itemsByStatus: Map<Boolean, List<GetTodosResponse.TodoItem>>,
-            val error: Boolean = false,
-        )
+import io.homeassistant.companion.android.util.compose.HomeAssistantGlanceTheme
+import io.homeassistant.companion.android.util.compose.HomeAssistantGlanceTypography
+import io.homeassistant.companion.android.util.compose.glance.glanceStringResource
+import io.homeassistant.companion.android.widgets.todo.TodoState.Companion.getColors
+
+/**
+ * Glance widget for managing and displaying a Todo List.
```

Comment:
I think we have different ideas of documenting code, but some of this feels like fluff / repeats the base Glance widget docs without adding much to it / should be in the developer docs instead of code probably (like: try to follow these guidelines).

Reply:
When writing it I was wondering if it was the place for this. So i'm going to move it. 
My initial idea was to find a way to make this todo widget a bit verbose on purpose to showcase and explain how things work. 
But I might have missed this goal.